src/App.js

```diff
@@ -22,17 +25,22 @@ function App() {
                 <WeatherButton />
                 <WeatherChart />
               </Route>
-              <Route exact path={'/work'}>
-                <WorkContextProvider>
-                  <WorkForm />
-                  <WorkTable />
-                </WorkContextProvider>
+              <Route exact path={WORK_KEY}>
+                <WorkDataContextProvider>
```

Comment:
I'd still suggest putting the providers outside the router in case you ever need to use a value from the provider.